Backpack/Snackbar/Classes/BPKSnackbar.m

```diff
@@ -0,0 +1,370 @@
+    self.actionButton.image = buttonIcon;
+    self.leftIconContainer.image = leftIcon;
+
+    if (buttonTitle != nil || buttonIcon != nil) {
+        self.actionButton.hidden = NO;
+        self.labelButtonSpacing.constant = BPKSpacingLg;
+        [self.actionButton setContentCompressionResistancePriority:751 forAxis:UILayoutConstraintAxisHorizontal];
+    } else {
+        self.actionButton.hidden = YES;
+        self.labelButtonSpacing.constant = 0;
+        [self.actionButton setContentCompressionResistancePriority:UILayoutPriorityDefaultHigh
+                                                           forAxis:UILayoutConstraintAxisHorizontal];
+    }
+
+    if (leftIcon != nil) {
+        self.leftIconContainer.hidden = NO;
+        self.labelLeadingSpacing.constant = BPKSpacingMd;
+    } else {
+        self.leftIconContainer.hidden = YES;
+        self.labelLeadingSpacing.constant = 0;
+    }
+
+    if (duration != BPKSnackbarDurationIndefinite) {
+        self.durationInMillis = [self durationInMilliseconds:duration];
+        if (self.durationInMillis > 0) {
+            self.timer = [NSTimer scheduledTimerWithTimeInterval:self.durationInMillis / 1000.0
+                                                          target:self
+                                                        selector:@selector(snackbarTimeOut)
+                                                        userInfo:nil
+                                                         repeats:NO];
+        }
+    }
+
+    [self updateLayout];
+}
+
+- (void)updateLayout {
+    UIView *presentingView = self.parentViewController.view;
+    [presentingView addSubview:self];
+
+    if (self.superview) {
+        [NSLayoutConstraint activateConstraints:@[
+            [self.leftAnchor constraintEqualToAnchor:self.superview.leftAnchor],
+            [self.rightAnchor constraintEqualToAnchor:self.superview.rightAnchor]
+        ]];
+
+        NSLayoutYAxisAnchor *bottomAnchor = self.superview.safeAreaLayoutGuide.bottomAnchor;
+
+        self.bottomConstraint =
+            [self.bottomAnchor constraintEqualToAnchor:bottomAnchor
+                                              constant:[self bottomInsetWithKeyboardHeight:self.keyboardHeight]];
+        self.bottomConstraint.active = YES;
+
+        self.heightConstraint = [self.heightAnchor constraintEqualToConstant:BPKSnackbarHeight];
+        self.heightConstraint.active = YES;
+
+        [self layoutIfNeeded];
+    }
+}
+
+- (void)show {
+    // First check if there's another snackbar presented
+    [self findAndRemoveSnackbars];
+
+    CGFloat distance = self.heightConstraint.constant;
+    self.snackbarViewTopConstraint.constant = distance;
+    [self layoutIfNeeded];
+
+    [UIView animateWithDuration:BPKSnackbarAnimationDuration
+                          delay:0
+                        options:UIViewAnimationOptionCurveEaseOut
+                     animations:^{
+                       self.snackbarViewTopConstraint.constant = 0;
```

Comment:
If the user has `reduceMotion` enabled we should probably fade it in and out instead of animating, but let's address that in another PR to avoid making this one even bigger!